app/models/king.rb

```diff
@@ -13,6 +17,35 @@ def valid_move?(x,y)
     return false
   end
 
+
+  def can_castle?(x,y)
+    updated_at == created_at && \
+    ((x == 6 && castle_rook && castle_rook.x_position == 7 && \
+    castle_rook.updated_at == castle_rook.created_at && \
+    castle_rook.path_blocked?(5, y_position) == false) || \
+    (x == 2 && castle_rook && castle_rook.x_position == 0 && \
+    castle_rook.updated_at == castle_rook.created_at && \
+    castle_rook.path_blocked?(3, y_position) == false )) && \
+    [y, castle_rook.y_position].uniq == y_position
+  end
+
+
+  def castle_rook
+    @castle_rook = game.pieces.where(x_position: 0, type: Rook).first || game.pieces.where(x_position: 7, type: Rook).first
+  end
+
+
+  def castle!
+    castle_rook.move_to((rook.x_position - (rook.x_position - king.x_position).abs + 1), y_position)
+  end
+
+
+  def move_to(x,y)
+    castle! if can_castle?(x,y)
```

Comment:
Since we have to ensure that the path for the rook is not blocked (nothing between the rook and the king), won't this cause an error if the king is moved before the rook?